apps/ledger-live-mobile/src/screens/Accounts/AccountRow.tsx

```diff
@@ -46,7 +46,7 @@ const AccountRow = ({
   useEnv("HIDE_EMPTY_TOKEN_ACCOUNTS");
   const currency = getAccountCurrency(account);
   const parentAccount = useSelector((state: State) =>
-    parentAccountSelector(state, { account: account as ChildAccount }),
+    parentAccountSelector(state, { account: account as TokenAccount }),
```

Comment:
Why force cast with a new type? The removed one should imply that the `account` cannot be cast, no?

Reply:
i'm not familiar with this specific code. 
I will create a ticket on Jira to investigate why we need this casting / if it can be fixed, but I think out of scope of the type simplification of this PR I think.  
The flaw is in the existing code of develop, it's just that here the new ChildAccount is now TokenAccount, it seems like we loosely expect account is a token account here and we need this cast to not break the typing.

Reply:
I mean also casting from `ChildAccount` (which is a specific type) to `TokenAccount` is quite weird as it is not the same type.

Reply:
in this PR, ChildAccount === TokenAccount

Reply:
ooh I see what you mean. i think it was a mistake in the past indeed, instead of ChildAccount we meant to do SubAccount.
because the parentAccountSelector takes a SubAccount out there. and the ChildAccount was type matching to be a SubAccount, the problem is that, indeed, casting hide the type issue in this place.

It isn't typed well at this selector side, it could accept an AccountLike so we get rid of this casting

```js
export const parentAccountSelector = createSelector(
  accountsSelector,
  (_: State, { account }: { account?: SubAccount }) => (account ? account.parentId : null),
  (accounts, accountId) => accounts.find(a => a.id === accountId),
);```

**i'm pushing a fix.**

*(side note: the usecase for such a selector is very weird, we shouldn't have to infer back the parent, it should be provided in the first place... but i'm not going to look at this now)*